lib/checkexceptionsafety.cpp

```diff
@@ -374,26 +374,27 @@ void CheckExceptionSafety::rethrowNoCurrentException()
 {
     logChecker("CheckExceptionSafety::rethrowNoCurrentException");
     const SymbolDatabase* const symbolDatabase = mTokenizer->getSymbolDatabase();
+    
     for (const Scope * scope : symbolDatabase->functionScopes) {
         const Function* function = scope->function;
         if (!function)
             continue;
 
-        // Rethrow can be used in 'exception dispatcher' idiom which is FP in such case
-        // https://isocpp.org/wiki/faq/exceptions#throw-without-an-object
-        // We check the beginning of the function with idiom pattern
-        if (Token::simpleMatch(function->functionScope->bodyStart->next(), "try { throw ; } catch ("))
-            continue;
-
+        //// Rethrow can be used in 'exception dispatcher' idiom which is FP in such case
+        //// https://isocpp.org/wiki/faq/exceptions#throw-without-an-object
+        //// 
+        //// After the idiom is found, continue to the next function scope in the above for loop.
         for (const Token *tok = function->functionScope->bodyStart->next();
-             tok != function->functionScope->bodyEnd; tok = tok->next()) {
+             tok != function->functionScope->bodyEnd && 
+             !Token::simpleMatch(tok, "try { throw ; } catch ("); tok = tok->next()) {
```

Comment:
hmm you could also put this condition in the loop body.

     if (Token::simpleMatch(tok, "try { throw ; } catch ("))
         break;

I don't have a really strong opinion but feel that it's imho a bit preferable since the loop condition will be easier.